### ISSUE TITLE
Trinity crashes during ENR lookup

### ISSUE THREAD

**Issue by carver** [ethereum/trinity#1586]
* Trinity version: 08c6784

### What is wrong?

```
Traceback (most recent call last):
  File "/usr/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/jcarver/code/trinity/trinity/extensibility/trio.py", line 56, in run_process
    trio.run(cls._do_run, boot_info)
  File "/home/jcarver/code/trinity/venv/lib/python3.6/site-packages/trio/_core/_run.py", line 1804, in run
    raise runner.main_task_outcome.error
  File "/home/jcarver/code/trinity/trinity/extensibility/trio.py", line 70, in _do_run
    nursery.cancel_scope.cancel()
  File "/home/jcarver/code/trinity/venv/lib/python3.6/site-packages/trio/_core/_run.py", line 730, in __aexit__
    raise combined_error_from_nursery
  File "/home/jcarver/code/trinity/trinity/components/builtin/peer_discovery/component.py", line 101, in do_run
    await async_service.run_trio_service(discovery_service)
  File "/home/jcarver/code/trinity/venv/lib/python3.6/site-packages/async_service/trio.py", line 156, in run_service
    await manager.run()
  File "/home/jcarver/code/trinity/venv/lib/python3.6/site-packages/async_service/trio.py", line 208, in run
    for _, exc_value, exc_tb in self._errors
  File "/home/jcarver/code/trinity/venv/lib/python3.6/site-packages/async_service/base.py", line 301, in _run_and_manage_task
    await task.run()
  File "/home/jcarver/code/trinity/venv/lib/python3.6/site-packages/async_service/trio.py", line 76, in run
    await self._async_fn(*self._async_fn_args)
  File "/home/jcarver/code/trinity/p2p/discovery.py", line 706, in receive
    await handler(node, payload, message_hash)
  File "/home/jcarver/code/trinity/p2p/discovery.py", line 860, in recv_enr_response
    existing_enr = self.node_db.get_enr(enr.node_id)
  File "/home/jcarver/code/trinity/p2p/discv5/enr_db.py", line 58, in get_enr
    return rlp.decode(self.db[self._get_enr_key(node_id)], sedes=ENR)
  File "/home/jcarver/code/py-evm/eth/db/backends/level.py", line 55, in __getitem__
    raise KeyError(key)
KeyError: b'\x94\xb8"`\xdb3\xe6=\x1bt%\xf1\x11R\xa5\x0b\x91\x88u\x91\xa2D{#1\x05\xf3\x92\xdb\xaer\xce:enr'
```

### How can it be fixed

cc @gsalgado

**Comment by gsalgado** [comment]
Did you get any logged warnings immediately before the crash?

**Comment by gsalgado** [comment]
Anyway... this will happen when we receive a message from a given node, using an address that is different from the one we have in our node DB. In that case the `receive()` method will delete the node DB entry to force a refresh. However, when the message is an ENR_RESPONSE, the handler was assuming there would be an existing entry in the node DB (which is fair because we only request ENRs from nodes we've bonded with)

I think the correct fix here is to have `receive()` replace the existing entry in the node DB with the new one. This will also be the correct approach once we fix #1451 and stop passing `Node` instances around -- instead we'll just be passing node IDs.

**Comment by carver** [comment]
> Did you get any logged warnings immediately before the crash?

I didn't see any.
